packages/presentation/lib/navigation/base_page.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/material.dart';
+
+import 'base_arguments.dart';
+
+class BasePage<T extends BaseArguments> extends Page {
+  const BasePage({
+    @required required LocalKey key,
+    @required required String name,
+    @required required this.builder,
+    T? arguments,
+  }) : super(key: key, name: name, arguments: arguments);
+
+  final WidgetBuilder builder;
+
+  @override
+  Route createRoute(BuildContext context) => _AppRoute(
+        settings: this,
+        builder: builder,
+      );
+}
+
+class _AppRoute extends MaterialPageRoute {
```

Comment:
probably there's no need in this class. u can use MaterialPageRoute directly instead of _AppRoute